app/src/main/java/com/realityexpander/tasky/presentation/agenda_screen/AgendaState.kt

```diff
@@ -0,0 +1,32 @@
+package com.realityexpander.tasky.presentation.agenda_screen
+
+import com.realityexpander.tasky.presentation.util.UiText
+
+data class AgendaState(
+    val username: String = "",
+    val email: String = "",
+    val password: String = "",
+    val confirmPassword: String = "",
+    val isPasswordVisible: Boolean = false,
+
+    val isLoading: Boolean = false,
+    val isLoggedIn: Boolean = false,
+    val isError: Boolean = false,
+
+    val isInvalidUsername: Boolean = false,
+    val isShowInvalidUsernameMessage: Boolean = false,
```

Comment:
Could it make sense to combine these into one field?
```
val invalidUsernameMessage: String? = null

// --------------------------------------

if(invalidUsernameMessage == null) {
    // No error and don't show message
} else {
    // Show message and there is an error
}
```

Reply:
I have them in two separate fields because on is for while the user is typing, and the other is after they attempt to login/register and I show them the "red error" message. I only want to show the "red error" once they have ignored the "invalid entry" icon in the field. 

So basically I am tracking 2 states: 
1. While the user types, a small icon will show the "check" or the "warning" icon.
2. When the user attempts to login/register, they will see the error messages below the fields, letting them know what is wrong and how to fix it.

Reply:
Is there a better way to handle this?